bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -174,6 +165,18 @@ protected synchronized void modified(Map<String, Object> properties) {
         }
     }
 
+    private void delayedInitialize(ScheduledExecutorService executor) {
+        // we wait until no further new links or items are announced in order to avoid creation of
+        // items which then must be removed again immediately.
+        final long diff = System.nanoTime() - lastUpdate - TimeUnit.SECONDS.toNanos(2);
+        if (diff < 0) {
```

Comment:
Hm, looking closer, I am afraid that logic won't work. We are waiting the two seconds to see if `lastUpdate`might change meanwhile, so we have to repeat this if statement over and over - at the moment, we would only check it once when "modified" is called and never again.
So I guess we have to stick to the separate thread after all (or re-schedule a job upon every update of `lastUpdate`).